app/controllers/sessions_controller.rb

```diff
@@ -24,4 +25,30 @@ def create
   def auth_hash
     request.env['omniauth.auth']
   end
+
+  def persist_services_from_session
+    return unless session[:services].present?
+  
+    session[:services].each do |service_data|
+      if service_data.respond_to?(:to_h)
+        service_data = service_data.to_h
+      end
+      
+      service = Service.find_or_initialize_by(id: service_data['id'])
+  
+      service.name = service_data['name']
+  
+      if service_data['service_at_locations'].present?
+        location_data = service_data['service_at_locations'].first&.dig('location')&.to_h
+  
+        if location_data
+          service.location_name = location_data['name']
+          service.address_1 = location_data['address_1']
+          service.city = location_data['city']
+          service.postal_code = location_data['postal_code']
+          service.save
```

Comment:
if we're storing location data we should store it in a table that matches a similar structure to Outpost. a locations endpoint or a way to filter an `api/v1/services` with multiple service id's could help with this as well.